### ISSUE TITLE
training with multiple losses

### ISSUE THREAD

**Issue by roebel** [keras-team/keras#20986]
Hello,

I frequently run into the same issue: I have a model with a single output and would like to use multiple weighted losses. This is quite common today for Neural Vocoders. As far as I see, this common use pattern appears problematic in Keras, as the compile function requires as many losses as outputs. This means that when I vary the number of losses, I need to change the number of outputs. 

Another alternative is to create a composed loss that internally uses the weights and calculates the combined loss. The problem is that I will not see the value of the individual losses as metrics. 

My question here is: Do  I miss something in the API, or is this common use case not supported at all, in which case, this would be a feature-request.

Many thanks for your help.

**Comment by edge7** [comment]
Hi,
In that case, I usually use a combined loss, and then I add the single losses in the metrics to check their single values.
Hope it makes sense

**Comment by roebel** [comment]
For info it works nicely with minor changes in the MeanParts Metric.

```python
class MeanParts(keras.metrics.Mean):
...
def mean_update_state(self, values, sample_weight=None, ignored=True):
      super().update_state(values, sample_weight)

def update_state(self, *arg, **kwargs):
      pass
```

Thanks again 
@edge7
